spectre_analyses/tasks.py

```diff
@@ -0,0 +1,140 @@
+from contextlib import contextmanager
+from functools import partial
+import json
+import os
+import pickle as pkl
+import shutil
+import sys
+
+import celery
+from celery.utils.log import get_task_logger
+
+import spdata.reader
+import spdata.types as ty
+
+from spectre_analyses.celery import app
+import matlab_hooks as mh
+
+
+FILESYSTEM_ROOT = os.path.abspath(os.sep)
+DATA_ROOT = os.path.join(FILESYSTEM_ROOT, 'data')
+STATUS_PATHS = {
+    'all': FILESYSTEM_ROOT,
+    'done': os.path.join(FILESYSTEM_ROOT, 'data'),
+    'processing': os.path.join(FILESYSTEM_ROOT, 'temp'),
+    'failed': os.path.join(FILESYSTEM_ROOT, 'failed')
+}
+Name = str
+Path = str
+
+
+def _data_path(dataset_name: Name) -> Path:
+    return os.path.join(DATA_ROOT, dataset_name, 'text_data', 'data.txt')
+
+
+def _load_data(dataset_name) -> ty.Dataset:
+    path = _data_path(dataset_name)
+    with open(path) as infile:
+        return spdata.reader.load_txt(infile)
+
+
+@contextmanager
+def _open_analysis(dataset_name: str, algorithm_name: str, analysis_name: str):
+    analysis_root = os.path.join(
+        STATUS_PATHS['processing'],
+        dataset_name,
+        algorithm_name,
+        analysis_name,
+    )
+    os.makedirs(analysis_root)
+    try:
+        yield analysis_root
+        dest_root = os.path.join(
+            STATUS_PATHS['failed'],
+            dataset_name,
+            algorithm_name,
+            analysis_name
+        )
+    except Exception as ex:
+        dest_root = os.path.join(
+            STATUS_PATHS['done'],
+            dataset_name,
+            algorithm_name,
+            analysis_name
+        )
+        raise RuntimeError() from ex
```

Comment:
You should move to `STATUS_PATHS['failed']` directory in case of exception. Now you do it the other way around.

Reply:
Good catch.